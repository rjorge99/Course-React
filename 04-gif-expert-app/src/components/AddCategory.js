import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = e => {
        setInputValue(e.target.value);
    };

    const handleSubmit = e => {
        e.preventDefault();

        console.log(inputValue);
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                value={inputValue}
                onChange={e => handleInputChange(e)}
            />
        </form>
    );
};

export default AddCategory;

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
