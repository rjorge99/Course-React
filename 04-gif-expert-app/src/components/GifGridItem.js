import React from 'react'

export const GifGridItem = ({ title, url }) => {
    
    return (
        <div>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
