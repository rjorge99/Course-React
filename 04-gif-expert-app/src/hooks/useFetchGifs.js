import { useEffect, useState } from 'react';
import { getGifs } from '../helpers.js/getGifs';

export const useFetchGifs = category => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category).then(img => {
            setState({
                data: img,
                loading: false
            });
        });
    }, [category]);

    return state;
};
