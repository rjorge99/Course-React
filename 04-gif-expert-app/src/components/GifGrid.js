import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <div>
            <h3>{category}</h3>
            {loading && <p>Loading...</p>}
            {images.map(img => (
                <GifGridItem key={img.id} {...img} />
            ))}
        </div>
    );
};
