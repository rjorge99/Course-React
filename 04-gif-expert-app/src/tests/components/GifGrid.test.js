import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prubas de GifGrid', () => {
    const category = 'Cat';

    test('Debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({ data: [], loading: true });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se carga imagenes', () => {
        const imgs = [{ id: 'abc', url: 'address', title: 'title' }];
        useFetchGifs.mockReturnValue({ data: imgs, loading: false });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    });
});
