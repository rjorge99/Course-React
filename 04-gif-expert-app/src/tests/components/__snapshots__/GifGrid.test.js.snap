// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas de GifGrid Debe mostrar el componente correctamente 1`] = `
<div>
  <h3>
    Cat
  </h3>
  <p>
    Loading...
  </p>
</div>
`;

exports[`Prubas de GifGrid Debe mostrar items cuando se carga imagenes 1`] = `
<div>
  <h3>
    Cat
  </h3>
  <GifGridItem
    id="abc"
    key="abc"
    title="title"
    url="address"
  />
</div>
`;
